lib/local-cluster/lib/Control/Monitoring/Server.hs

```diff
@@ -0,0 +1,87 @@
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE RankNTypes #-}
+
+-- | How to run a monitor
+module Control.Monitoring.Server
+    ( Protocol (..)
+    , runTCPControl
+    )
+where
+
+import Prelude
+
+import Control.Monad
+    ( forever
+    , (<=<)
+    )
+import Control.Monitoring.Monitor
+    ( Monitor (..)
+    )
+import Network.Simple.TCP
+    ( HostPreference (HostAny)
+    , serve
+    )
+import Network.Socket
+    ( socketToHandle
+    )
+import System.IO
+    ( hGetLine
+    , hPutStrLn
+    )
+import Text.Read
+    ( readMaybe
+    )
+import UnliftIO
+    ( IOMode (..)
+    , MonadUnliftIO
+    , UnliftIO (..)
+    , askUnliftIO
+    , async
+    , liftIO
+    , link
+    )
+
+-- | The command you can send to the monitor
+data Protocol
+    = Pull -- ^ Pull the next trace, if in pulling state
+    | Switch -- ^ Switch between pulling and not-pulling or vice versa
+    | Observe -- ^ Observe the current state
+    | Kill -- ^ Try to kill the program on the next trace
+    deriving stock (Show, Read)
+
+-- | Run a TCP control of the monitor on a given port
+-- We use CPS even if it's not needed in this implementation as a
+-- future-proofing measure.
+runTCPControl
```

Comment:
```suggestion
serveMonitorOnTCP
```

I think that this name better describes what the function does: It serves an API (here: the `Monitor` record) on a protocol (here `TCP`). In that sense, this function is similar to `serveWallet`.